### DISCUSSION TITLE
The Need for the Existence of UsingUnsafe.OfFactory

### DISCUSSION THREAD

**Discussion by githubcheng2978** [apache/skywalking#12252] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

agentBuilder.with(new AgentBuilder.InjectionStrategy.UsingUnsafe.OfFactory(factory)); 
Is there a need for this code to be there, because bootstrap-plugins minions work fine when I remove it!

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
You may need to ask from Bytebuddy for the final answer about this. From my understanding, this should be able to adopt the gap before and after JDK9.

**An injection strategy that uses sun. misc. Unsafe or jdk. internal. misc. Unsafe to inject classes.**

**Comment by dsc6636926** [comment]
https://github.com/apache/skywalking-java/blob/ffbd90c3d371259d44cca038f3810a99c9de7594/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java#L119

this line may change InjectionStrategy from InjectionStrategy.UsingReflection to InjectionStrategy.UsingUnsafe ,this means when i use bootstrap plugins ,the InjectionStrategy is InjectionStrategy.UsingUnsafe。is this correct?

**Reply by dsc6636926** [reply]
InjectionStrategy in agentBuilder is global
